.buildkite/scripts/cloud-cleanup.sh

```diff
@@ -0,0 +1,188 @@
+        return 1
+    fi
+    return 0
+}
+
+cloud_reaper_aws() {
+    echo "Validating configuration"
+    docker run --rm -v $(pwd)/.buildkite/configs/cleanup.aws.yml:/etc/cloud-reaper/config.yml \
+      -e ACCOUNT_SECRET="${ELASTIC_PACKAGE_AWS_SECRET_KEY}" \
+      -e ACCOUNT_KEY="${ELASTIC_PACKAGE_AWS_ACCESS_KEY}" \
+      -e ACCOUNT_PROJECT="${ELASTIC_PACKAGE_AWS_USER_SECRET}" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --config /etc/cloud-reaper/config.yml \
+          validate
+
+    echo "Scanning resources"
+    docker run --rm -v $(pwd)/.buildkite/configs/cleanup.aws.yml:/etc/cloud-reaper/config.yml \
+      -e ACCOUNT_SECRET="${ELASTIC_PACKAGE_AWS_SECRET_KEY}" \
+      -e ACCOUNT_KEY="${ELASTIC_PACKAGE_AWS_ACCESS_KEY}" \
+      -e ACCOUNT_PROJECT="${ELASTIC_PACKAGE_AWS_USER_SECRET}" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --config /etc/cloud-reaper/config.yml \
+          ${COMMAND} | tee "${AWS_RESOURCES_FILE}"
+}
+
+cloud_reaper_gcp() {
+    echo "Validating configuration"
+    docker run --rm -v $(pwd)/.buildkite/configs/cleanup.gcp.yml:/etc/cloud-reaper/config.yml \
+      -e ACCOUNT_SECRET="${ELASTIC_PACKAGE_GCP_KEY_SECRET}" \
+      -e ACCOUNT_KEY="${ELASTIC_PACKAGE_GCP_EMAIL_SECRET}" \
+      -e ACCOUNT_PROJECT="${ELASTIC_PACKAGE_GCP_PROJECT_SECRET}" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --config /etc/cloud-reaper/config.yml \
+          validate
+
+    echo "Scanning resources"
+    docker run --rm -v $(pwd)/.buildkite/configs/cleanup.gcp.yml:/etc/cloud-reaper/config.yml \
+      -e ACCOUNT_SECRET="${ELASTIC_PACKAGE_GCP_KEY_SECRET}" \
+      -e ACCOUNT_KEY="${ELASTIC_PACKAGE_GCP_EMAIL_SECRET}" \
+      -e ACCOUNT_PROJECT="${ELASTIC_PACKAGE_GCP_PROJECT_SECRET}" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --config /etc/cloud-reaper/config.yml \
+          ${COMMAND} | tee "${GCP_RESOURCES_FILE}"
+}
+
+echo "--- Cleaning up GCP resources older than ${DELETE_RESOURCES_BEFORE_DATE}..."
+cloud_reaper_gcp
+
+if any_resources_to_delete "${GCP_RESOURCES_FILE}"; then
+    echo "Pending GCP resources"
+    resources_to_delete=1
+fi
+
+echo "--- Cleaning up AWS resources older than ${DELETE_RESOURCES_BEFORE_DATE}..."
+cloud_reaper_aws
+
+if any_resources_to_delete "${AWS_RESOURCES_FILE}" ; then
+    echo "Pending AWS resources"
+    resources_to_delete=1
+fi
+
+if [ "${resources_to_delete}" -eq 1 ]; then
+    message="There are resources to be deleted"
+    echo "${message}"
+    if running_on_buildkite ; then
+         buildkite-agent annotate \
+             "${message}" \
+             --context "ctx-cloud-reaper-error" \
+             --style "error"
+    fi
+    exit 1
```

Comment:
Do we want to exit here? We could continue and check other resources, right?

Reply:
Thanks, I forgot to remove this exit!
I've updated the code to allow running all steps and just exit with failure at the end.